mne/transforms.py

```diff
@@ -1313,26 +1312,43 @@ def _find_vector_rotation(a, b):
 
 
 @jit()
-def _fit_matched_points(p, x):
+def _fit_matched_points(p, x, weights=None, scale=False):
     """Fit matched points using an analytical formula."""
     # Follow notation of P.J. Besl and N.D. McKay, A Method for
     # Registration of 3-D Shapes, IEEE Trans. Patt. Anal. Machine Intell., 14,
     # 239 - 255, 1992.
     #
+    # The original method is actually by Horn, Closed-form solution of absolute
+    # orientation using unit quaternions, J Opt. Soc. Amer. A vol 4 no 4
+    # pp 629-642, Apr. 1987. This paper describes how weights can be
+    # incorporated (by modifying centroids and )
+    #
     # Caution: This can be dangerous if there are 3 points, or 4 points in
     #          a symmetric layout, as the geometry can be explained
     #          equivalently under 180 degree rotations.
     #
-    # XXX eventually we should use this in coreg to speed things up.
+    # Eventually this can be extended to also handle a uniform scale factor,
+    # as well.
     assert p.shape == x.shape
     assert p.ndim == 2
     assert p.shape[1] == 3
-    mu_p = mean(p, axis=0)  # eq 23
-    mu_x = mean(x, axis=0)
-    Sigma_px = np.dot(p.T, x) / p.shape[0] - np.outer(mu_p, mu_x)  # eq 24
+    # (weighted) centroids
```

Comment:
Maybe you can use np.average and the accepts weights ?

Reply:
This is in `jit`-ed code, and Numba does not support it. So not worth the speed loss :)